awscrt/auth.py

```diff
@@ -306,7 +319,7 @@ def _on_complete(error_code, binding):
         try:
             _awscrt.credentials_provider_get_credentials(self._binding, _on_complete)
         except Exception as e:
-            future.set_result(e)
+            future.set_exception(e)
```

Comment:
good catch

Reply:
oye, this bug has been there since forever. I guess I never tested the failure case until now 😬